modules/fedimint-ln-client/src/lib.rs

```diff
@@ -580,6 +663,50 @@ impl LightningClientModule {
         ))
     }
 
+    /// Create an output that funds an incoming contract within the federation
+    /// This directly completes a transaction between users, without involving a
+    /// gateway
+    pub async fn create_incoming_output(
+        &self,
+        operation_id: OperationId,
+        invoice: Invoice,
+    ) -> anyhow::Result<(
+        ClientOutput<LightningOutput, LightningClientStateMachines>,
+        ContractId,
+    )> {
+        let payment_hash = invoice.payment_hash();
+        let invoice_amount = Amount {
+            msats: invoice
+                .amount_milli_satoshis()
+                .ok_or(InternalTxError::AmountError)?,
+        };
+
+        let (incoming_output, contract_id) = create_incoming_contract_output(
+            &self.module_api,
+            *payment_hash,
+            invoice_amount,
+            self.redeem_key,
+        )
+        .await?;
+
+        let client_output = ClientOutput::<LightningOutput, LightningClientStateMachines> {
```

Comment:
Hmmm why was this part not included in `create_incoming_contract_output`? Its the same for the gateway and the client right?

Reply:
What's the difference between this function and `create_funding_incoming_contract_output` in the gateway?

Reply:
No, they differ on the typed state machine wrap within the client output. These differ in type only, where here we return a `LightningClientStateMachines::InternalPay`, the gateway scenario returns a `GatewayClientStateMachines::Receive`.

In my opinion, these state machines each make sense for their respective scenarios, so there is no need to find some universal name for a state machine we can return in both scenarios, from a shared utility fn